VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +122,110 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    public static class DiscussionResponsesLoader implements
+            InfiniteScrollUtils.PageLoader<DiscussionComment> {
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private Context context;
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private int nextPage = 1;
+        private String threadId;
+        private boolean isQuestionTypeThread;
+
+        private ThreadComments unendorsedComments;
+        private boolean isEndorsedFetched = false;
+
+        public DiscussionResponsesLoader(@NonNull Context context,@NonNull String threadId,
+                                         boolean isQuestionTypeThread) {
+            this.context = context;
+            this.threadId = threadId;
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        @Override
+        public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) {
+                getEndorsedList();
+            }
+            getCommentList();
+
+        }
+
+        /**
+         * Gets the list of endorsed answers for a {@link DiscussionThread.ThreadType#QUESTION}
+         * type discussion thread
+         */
+        protected void getEndorsedList() {
+            if (getEndorsedListTask != null) {
+                getEndorsedListTask.cancel(true);
+            }
+            getEndorsedListTask = new GetCommentListTask(context, threadId, 1, isQuestionTypeThread,
+                    true) {
+                @Override
+                public void onSuccess(ThreadComments threadComments) {
+                    isEndorsedFetched = true;
+                    callback.onPartialPageLoaded(threadComments.getResults());
```

Comment:
The `onSuccess()` callback should be guarded with the `callback != null` condition, as it is possible for the result to be delivered even after task cancellation.